lib/source-code/source-code.js

```diff
@@ -193,6 +193,16 @@ class SourceCode extends TokenStore {
         Object.freeze(this.lines);
     }
 
+    /**
+     * Removes inline comments (`\/*   *\/`) from specified text.
+     * @param {string} text Text to process.
+     * @returns {string} Text without inline comments.
+     * @public
+     */
+    static removeInlineComments(text) {
+        return text.replace(/\/\*.*?\*\//gu, "");
```

Comment:
I don't think this it's a good idea to add this to the public API. This check only works when given the text between two adjacent tokens. If allowed any arbitrary text, a string containing this pattern could be mutated.

Reply:
As stated in name and comment this method is intended to remove _inline_ comments i.e. `/*   */` comments inside a line. So it is quite generic and can be helpful (at least in this module).

But I can replace the method by private function.

Reply:
I understand. However, this would also mutate the following code:

```
const constructComment = `/*${insertCommentText()}*/`;
```
becomes
```
const constructComment = ``;
```

I actually think the code this is being lifted from in `isSpaceBetweenTokens` is also buggy, and there is a [PR](https://github.com/eslint/eslint/pull/12491) open to fix it.